Formula/boost.rb

```diff
@@ -42,8 +42,8 @@ class Boost < Formula
   # fix error: no member named 'make_array' in namespace 'boost::serialization'
   # https://svn.boost.org/trac/boost/ticket/12978
   patch :p2 do
-    url "https://github.com/boostorg/serialization/commit/1d86261.diff"
-    sha256 "155f603a00975a1702808be072c1420964feac8323de39c111a9d3a363a4ed9a"
+    url "https://github.com/boostorg/serialization/commit/1d8626158.diff?full_index=1"
```

Comment:
@mistydemeo I'm not convinced that we need the full commit hash for these, but since this is our first use of `?full_index=1`, it does seem fitting, so I've updated the PR accordingly :)